### PR TITLE
Filter device selector

### PR DESCRIPTION
**Pull request by alfredo-dotcms**

### CODE REVIEW

File: src/app/portlets/dot-edit-page/content/dot-edit-content.component.spec.ts [hunk 1]
```diff
@@ -788,7 +788,7 @@ describe('DotEditContentComponent', () => {
         });
     });
 
-    describe('Auto save', () => {
+    fdescribe('Auto save', () => {
```
Comment (fmontes) [inline src/app/portlets/dot-edit-page/content/dot-edit-content.component.spec.ts]: `f` here.

File: src/app/portlets/dot-edit-page/content/components/dot-edit-content-view-as-toolbar/dot-edit-content-view-as-toolbar.component.spec.ts [hunk 2]
```diff
@@ -227,7 +227,7 @@ describe('DotEditContentViewAsToolbarComponent', () => {
             expect(whatsChanged).toBe(null);
         });
 
-        describe('events', () => {
+        fdescribe('events', () => {
```
Comment (fmontes) [inline src/app/portlets/dot-edit-page/content/components/dot-edit-content-view-as-toolbar/dot-edit-content-view-as-toolbar.component.spec.ts]: `f` here.

File: src/app/view/components/dot-device-selector/dot-device-selector.component.ts [hunk 3]
```diff
@@ -21,14 +21,19 @@ export class DotDeviceSelectorComponent implements OnInit {
...
-                    .get()
-                    .pipe(
-                        map((devices: DotDevice[]) =>
-                            this.setOptions(this.dotMessageService.get('editpage.viewas.default.device'), devices)
+            .pipe(
+                mergeMap(() =>
+                    this.dotDevicesService
+                        .get()
+                        .pipe(
+                            flatMap((devices: DotDevice[]) => devices),
```
Comment (freddyucv) [inline src/app/view/components/dot-device-selector/dot-device-selector.component.ts]: Are you sure that we need a flatMap here?

File: src/app/view/components/dot-device-selector/dot-device-selector.component.ts [hunk 4]
```diff
@@ -21,14 +21,19 @@ export class DotDeviceSelectorComponent implements OnInit {
...
+                            filter((device: DotDevice) => +device.cssHeight > 0 && +device.cssWidth > 0),
+                            toArray(),
+                            map((devices: DotDevice[]) =>
+                                this.setOptions(this.dotMessageService.get('editpage.viewas.default.device'), devices)
+                            )
```
Comment (freddyucv) [inline src/app/view/components/dot-device-selector/dot-device-selector.component.ts]: I think it shoud be a tap

File: src/app/test/dot-device.mock.ts [hunk 5]
```diff
@@ -1,8 +1,16 @@
...
+        name: 'iphone',
+        inode: '1'
+    },
+    {
+        cssHeight: '0',
+        cssWidth: '0',
+        name: 'bad device',
+        inode: '2'
+    }
+];
```
Comment (freddyucv) [inline src/app/test/dot-device.mock.ts]: shouldn't we have one with cssHeight and/or cssWidth equals to null?

Comment (alfredo-dotcms) [reply src/app/test/dot-device.mock.ts]: If it's null, empty string or a zero (string or number) if will be casted to 0 (number). So not sure if needed!

File: src/app/view/components/dot-device-selector/dot-device-selector.component.ts (hunk 3 above)
Comment (alfredo-dotcms) [reply src/app/view/components/dot-device-selector/dot-device-selector.component.ts]: We need to convert the data so the filter operator can work with the data correctly

File: src/app/test/dot-device.mock.ts (hunk 5 above)
Comment (fmontes) [reply src/app/test/dot-device.mock.ts]: What is coming from the server when those fields are empty?

Comment (alfredo-dotcms) [reply src/app/test/dot-device.mock.ts]: empty string

File: src/app/view/components/dot-device-selector/dot-device-selector.component.ts (hunk 4 above)
Comment (alfredo-dotcms) [reply src/app/view/components/dot-device-selector/dot-device-selector.component.ts]: no, we need to return the options

File: src/app/test/dot-device.mock.ts (hunk 5 above)
Comment (fmontes) [reply src/app/test/dot-device.mock.ts]: Ok, I think we should mock that.

**Pull request merged**